src/circuit_proof/verifier.rs

```diff
@@ -156,11 +153,19 @@ impl<'a> VerifierCS<'a> {
         use util;
 
         let n = self.num_vars;
-        if generators.gens_capacity < n {
+        if self.generators.gens_capacity < n {
             return Err(R1CSError::InvalidGeneratorsLength);
         }
         // We are performing a single-party circuit proof, so party index is 0.
-        let gens = generators.share(0);
+        let gens = self.generators.share(0);
+
+        // Create a `TranscriptRng` from the transcript
+        let mut rng = {
+            let ctor = self.transcript.fork_transcript();
+
+            use rand::thread_rng;
+            ctor.reseed_from_rng(&mut thread_rng())
+        };
```

Comment:
nit: I think this could be slightly cleaner as
```
use rand::thread_rng;
let mut rng = self.transcript.fork_transcript().reseed_from_rng(&mut thread_rng());
```
The prover's code is more complicated, so it bundles stuff up into a constructor scope, but I don't think that that's needed here.

Reply:
👍